py/Interface.cpp

```diff
@@ -311,15 +312,23 @@ class PyDsrcArchiveRecordsReader
 
 class PyDsrcReadInMemory {
 	public:
-		void Open(const std::string& inDsrcFilename_) {
+		PyDsrcReadInMemory & Open(const std::string& inDsrcFilename_) {
 			dsrcInMemory = new DsrcInMemory(inDsrcFilename_);
 			TCheckPtrError(dsrcInMemory);
+			return *this;
 		}
 
 		void Close() {
 			delete dsrcInMemory;
 			TCheckPtrError(dsrcInMemory);
 			dsrcInMemory = NULL;
+			/**
+			 * Create an empty queue<string> and swap it with `chunk` for an O(1)
+			 * operation to empty the queue of lines that were last read from the
+			 * `.dsrc` file
+			 */
+			std::queue<std::string> empty;
+			std::swap(chunk, empty);
```

Comment:
also an alternative is to use `std::queue<std::string>().swap(chunk)` (see [here](https://stackoverflow.com/questions/709146/how-do-i-clear-the-stdqueue-efficiently/709161#comment4574774_709161)) assuming this requires > C++11
I'm not sure if there would be a performance difference rather than using `std::queue` (or maybe they are just aliases of one each other)

Reply:
Noting a relevant discussion made (here)[https://stackoverflow.com/questions/42718327/why-does-every-stl-container-have-a-swap-function-defined-as-a-member-function], and from my understanding there is no major performance difference nowadays especially for >= C++11